pandas/tests/io/test_stata.py

```diff
@@ -361,16 +361,20 @@ def test_encoding(self, version):
 
         # GH 4626, proper encoding handling
         raw = read_stata(self.dta_encoding)
-        encoded = read_stata(self.dta_encoding, encoding="latin-1")
+        with warnings.catch_warnings(record=True) as w:
+            encoded = read_stata(self.dta_encoding, encoding='latin-1')
+            assert len(w) == 1
```

Comment:
Hmmm...I was thinking for that you use `tm.assert_produces_warning` or `pytest.warns`.